crates/oxc_linter/src/rules/oxc/only_used_in_recursion.rs

```diff
@@ -279,19 +266,13 @@ fn create_diagnostic_jsx(
 }
 
 fn used_with_spread_attribute(node_id: NodeId, ctx: &LintContext) -> bool {
-    let opening_element = ctx
-        .nodes()
-        .ancestors(node_id)
-        .find_map(|node| match ctx.nodes().kind(node) {
-            AstKind::JSXOpeningElement(opening_element) => Some(opening_element),
-            _ => None,
-        })
-        .expect("should be safe");
-
-    opening_element
-        .attributes
-        .iter()
-        .any(|attr| matches!(attr, JSXAttributeItem::SpreadAttribute(_)))
+    ctx.nodes().iter_parents(node_id).any(|node| match node.kind() {
```

Comment:
after https://github.com/oxc-project/oxc/pull/7217 merges, this can be rewritten as:

```suggestion
    ctx.nodes().ancestor_kinds(node_id).any(|kind| match kind {
```

Reply:
The fix has been applied. Commit: be722431437790e1a0dba367f5458abf8b5f096b